core/lib/spree/core/controller_helpers/common.rb

```diff
@@ -73,8 +73,22 @@ def set_user_language
         #
         # Default layout is: +app/views/spree/layouts/spree_application+
         #
+
         def get_layout
-          Spree::Config[:layout]
+          layout = Spree::Config.layout
+
+          case layout
+          when String
+            layout
+          when Symbol
+            send(layout)
+          when Proc
+            Spree::Config.layout.call(self)
+          when true
+            raise ArgumentError, 'Layouts must be specified as a String, Symbol, Proc, false, or nil'
+          when nil
+            'spree/layouts/spree_application'
+          end
```

Comment:
My initial thought was to get rid of the `get_layout` method and just call `Spree::Config.layout` directly on line 15. This kinda worked when I was testing it manually. But I did not get the tests to pass like this method. No matter what I would set for the layout in the config file, it wouldn't take effect, maybe I was missing something?

Reply:
Anyways, this works without having to deprecate the layout preference or breaking any current applications

Reply:
@devchris I think the reason that it's not working in tests is that the controller has already been loaded by the time you attempt to stub the preference, and the `layout` call has already been made. This is usually resolved either by reloading the class (i.e., unloading the constant associated to the file and `load`ing it again) or by moving the logic into a method (which is what you've done).

Reply:
@aldesantis This seems to be a little bit more tricky than just simply "reloading" it in this case. But regardless, keeping it in a method gives us some flexibility when we decide to go the route to deprecate 'spree_application.html.erb' 😄

Reply:
Makes sense!